### PR TITLE
Update comments

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.Common/Helpers/AnalysisContextExtensions.cs [hunk 1]
```diff
@@ -71,7 +71,7 @@ private static void ReportDiagnostic(ReportingContext reportingContext)
                 return;
             }
 
-            // ... but for compatibility purposes we need to keep handling the old-fashioned way
+            // ... but for compatibility purposes we need to keep handling the old-fashioned way. Old SonarLint can be used with latest NuGet.
```
Comment (pavel-mikula-sonarsource) [inline analyzers/src/SonarAnalyzer.Common/Helpers/AnalysisContextExtensions.cs:74]: This took some effort to find if it's still valid or not.

File: analyzers/src/SonarAnalyzer.Common/SonarAnalyzer.Common.csproj [hunk 2]
```diff
@@ -65,10 +65,6 @@
     <AdditionalFiles Include="..\..\.sonarlint\sonaranalyzer-dotnet\CSharp\SonarLint.xml" Link="Properties\SonarLint.xml" />
   </ItemGroup>
 
-  <ItemGroup>
-    <Folder Include="Di\" />
-  </ItemGroup>
-
```
Comment (pavel-mikula-sonarsource) [inline analyzers/src/SonarAnalyzer.Common/SonarAnalyzer.Common.csproj:71]: Cleanup a mess that I unintentionally introduced in #4030

File: sonar-dotnet-shared-library/src/main/java/org/sonar/plugins/dotnet/tests/CoverageReportImportSensor.java [hunk 3]
```diff
@@ -108,7 +108,6 @@ void analyze(SensorContext context, Coverage coverage) {
       } else if (inputFile.type().equals(Type.TEST)) {
         fileCountStatistics.test++;
         LOG.debug("Skipping '{}' as it is a test file.", filePath);
-        // Do not log for test files to avoid pointless noise
       } else if (!coverageConf.languageKey().equals(inputFile.language())) {
```
Comment (pavel-mikula-sonarsource) [inline sonar-dotnet-shared-library/src/main/java/org/sonar/plugins/dotnet/tests/CoverageReportImportSensor.java:111]: This comment is out of date. The logging was added in #2874 but the old comment wasn't removed.

**Review by andrei-epure-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**